float64 ang0
float64 ang1
float64 ang2
float64 ang3
float64 ang4
float64 ang5
